### DISCUSSION TITLE
Can no longer monitor my generator

### DISCUSSION THREAD

**Discussion by Gadgetmann133** [jgyates/genmon#1068] [General]
I had everything set up using a Raspberry Pi and HAT and also using a Ethernet to USB device instead of WiFi.  It worked great and was able to monitor my Generac.  Using genmon.local:8000 URL which I think is correct?  It does not connect.  I am at home on my Wireless router on my laptop(rebooted a few times also) and I have link lights on the Ethernet to USB device.
Only thing that changed is my battery died I replaced it and am able to start the Generac.  
Is there a reset or something else I am missing to be able to connect to my Pi again?

**Comment by skipfire** [comment]
1. genmon:8000 is the default address (genmon.local is likely a resolvable name for it on most home networks) for the pre-loaded units from PintSize.  If you did your own setup then the default from the Raspberry Pi imager would be raspberrypi:8000, or by IP.
2. Did you disconnect the Pi and HAT before replacing the battery?  If so, possibly a connection was missed after replacing it.
3. Did you have notifications working before changing the battery?  If so, did you get the startup notifications after changing the battery?
4. What Pi do you have, and what does the LED on the Pi show?
5. If the HAT is a PintSize HAT, is the LED on it dark, faint blue, or a bright color?
6. How are you powering the Pi & HAT?
7. What controller does your generator have?  Evo, Evo2, Nexus?
8. Have you tried pinging the Pi?

Rebooting (or power cycling) is always a good troubleshooting step.  If the HAT is a PintSize HAT you need to remove the Molex cable as well as power cables to ensure it drops all power.

**Comment by jgyates** [comment]
In addtion to @skipfire's questions I have these questions:

1) do you have email setup on the pi? If you do, if you reboot your pi system do you receive the email message stating that genmon is starting? if you do is the IP address in the email the same as the one you are trying to access genmon with in your browser?

**Comment by Gadgetmann133** [comment]
Sorry for the delay getting back for your responses.
1.  After I installed everything I was able to connect using genmon.local:8000
2.  I did not disconnect it.
3.  Yes, notifications and test email/texts worked before.
4.  I believe it is a 3.
5. It is a pintsize HAT and shows bright blue led.
6. Powered from the battery with a cable from Pintsize.
7. Evo1 I am told, Model # G0070431, 22KW about 3 yrs old.
8.  I have not tried pining the Pi yet, however
9. I plugged my laptop into an ethernet cable and was able to connect to the Pi and generator, so I am ok with that.

Is there a procedure to reboot the Pi system?  
As noted, I had email setup before and it worked.
The odd thing looking at the logs, last August there was a Charger Missing AC: alarm code 0000 and also Battery problem alarm codes and also RPM Sensor alarm codes-1505.
I am wondering why I did not get any of those alarms via email as I had the email working?

**Reply by skipfire** [reply]
If you were able to connect once your laptop was hardwired it isn't a problem with the Pi, you probably need to look at your WiFi access point configuration.  Many WAPs/routers have security settings that can isolate wireless traffic.

**Reply by Gadgetmann133** [reply]
Ok, I will check my WAP settings, thanks.

**Comment by Gadgetmann133** [comment]
Another update......on email.
When I installed it I had the email working and after I hooked up ethernet to my laptop and connected to the Pi and did and email test, it failed. 
I get:
An error occurred:
Error initializing SMTP library:
[Errno 101]  Network is unreachable.
Please try again

My settings are:
Email account    *******@gmail.com
Email password   **********
Sender Address   ******@gmail.com
Sender name     Genmon
SMTP Server        mail.google.com
SMTP Server port    465

I searched this website for gmail settings but I did not find specific settings for gmail.

I must have incorrect settings, any ideas?  thanks

**Reply by jgyates** [reply]
Here is info on how to setup gmail: 

     https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon#setup-gmail-for-use-with-genmon

The network unreachable is due to your SMTP server or port being either invalid or unreachable. You should be using the settings listed in the above link. If it was working before with wifi (I assume that is how the pi was connected before you laptop?). If you connect Ethernet to the pi then that is the preferred network connection for the pi so you email request may be routed to your laptop instead of to the internet. Try using a ssh connection from your laptop to your pi (which is connected to wifi I assume) to log into to the pi via wifi.

**Reply by Jkh62** [reply]
Outbound Emails from Genmon

Sign-in to the Google account you want to use on a PC browser

Select your account picture and click on Manage you Google Account

Click the Security tab, then scroll down and click on 2-Step Verification

Turn it on if it is not already on (no way around this, if you really can't use MFA then stop here)

Scroll down to the bottom and click on App passwords

Select an app (Other is good) and give it a name (Genmon is good, or gmail SMTP is good, you can use this password for all IoT devices that need to send emails, not just Genmon)

Click on Generate and copy the password to your clipboard, or write it down (even though it tells you not to write it down, do it anyway, you will need to put it in your Genmon Settings

Log into Genmon and click on the Settings link

Scroll down to Outbound Email Settings and turn it on

Enter the email account you are using, the one you generated the application password in

Enter the application password you generated in your Google account settings

Sender address can be no-reply@gmail.com or you can put in your email if you want, if you have trouble getting emails delivered to your inbox directly, try putting your email address as the sender

Sender name can be whatever you want, Genmon is fine

SMTP server is smpt.gmail.com
Leave port at 587

Click the Test Email Settings button, you should receive a test email
Click the Save link at the bottom of the page, Genmon will restart and you should get an email in a couple minutes saying that Genmon was started

**Comment by Gadgetmann133** [comment]
I have tried the settings in that link you attached and it does not work.  When it worked before this was my setup:
I was able to connect to the Pi via WiFi from my laptop.  The Pi has ethernet run to it and I am using a Ethernet to USB adapter.
I have always used Ethernet to the Pi and the only thing that is changed now is I have to use Ethernet to my laptop to connect to the Pi(Genmon).

**Reply by jgyates** [reply]
From the details that I have, it sounds like your email setup if failing because you don't have a network path to the internet from the pi (hence the "Network is unreachable" error). When you plug in your laptop directly into the pi via and Ethernet cable, your pi is trying to access the internet though your laptop, not the wifi. When your pi tries to access the internet through your laptop it fails as your laptop is not a router or it is not in bridge mode. If your pi is connected to your wifi, then connect your laptop to your wifi also to access it, then configure you email. Anytime a device has two network connections the device is configured to use only one connection for it's main traffic flow, otherwise traffic will be circulated in a loop, causing errors and network traffic overload. Most computers default priority to a wired connection like Ethernet instead of Wifi. 

My recommendations are:
Do not connect your pi directly to your laptop. This is not a genmon issues, but rather an issue with how your are trying to connect your home network.
You can connect your pi to your local wifi **or** to an Ethernet switch or router. You can connect your laptop to your local wifi **or** to a Ethernet switch or router. If you connect either the pi or laptop directly to each other you have to take extra network steps to allow the traffic to be bridged or routed properly.

You can connect your pi to your wired Ethernet network, just not directly to your laptop, it has to have a path to your router. This is typically done by a switch connected to your router. then multiple items connect to the switch.

**Comment by Gadgetmann133** [comment]
Update and my setup:
My laptop is not directly hooked up to my Pi.
My Pi is not connected to WiFi.
I have Ethernet running to my Pi that goes into my wireless router.
My laptop used to connect to genmon via WiFi but does not but when I hook it up to Ethernet it works.
Plot twist:  I tried another laptop and my Ipad via WiFi and it connects to genmon.
I have tried the smtp.gmail.com  port 587   Error sending 534 application specific password p=InvalidSecondFactor e33   
I am have 2 factor enabled on my gmail.
I have tried mail.google.com  port 465     Error  initializing SMTP library:
[Errno 101] Network is unreachable.
Please try again

So as I first noted this all started because my battery was dead which I have replaced.  The battery died because the charging circuit is doa.
Charger missing AC  Alarm code 0000
RPM sensor Alarm code 1505

I replaced the T1 fuse in the transfer switch but that is not the problem.  I have 120V at the T1 wire on the block in the generator so I am told my Controller is DOA as that is where the internal battery charger circuit is.  
So I am about ready to scrap this entire SS, the genmon project email and the Generac too.

**Reply by jgyates** [reply]
I don't know how long you have been running genmon, but google changed some of their security [requirements in 2022](https://github.com/jgyates/genmon/issues/682) so unless you have been getting emails in the past it is possible that your email stopped working earlier (just a guess). Port 465 is an older less secure method and most email providers (apple, google, microsoft, etc) are trying to phase out less secure methods, so if they have not stopped using it, they probably will soon.

Unless you have some specific reason, I would not use port 465, it appears that the address mail.google.com is not accepting that port based on the error. The message "Error sending application specific password" indicates to me that you don't have your gmail account setup per the instructions. gmail works for me and I now a lot of folks use it. It is the most popular mail account used with genmon.

Another thing to keep in mind, if you do use port 465, it requires different genmon settings as the server will encrypt the email traffic differently. This is how you should have your genmon mail settings for smtp.gmail.com port 587
<img width="615" alt="Screenshot 2024-03-07 at 11 18 26 PM" 
src="https://github.com/jgyates/genmon/assets/4513286/676f6b8d-9fe3-49d1-8513-f5752f94a963">

**Comment by Gadgetmann133** [comment]
I configured my genmon as you have noted above.
Error message:
An error occurred:
Error initializing SMTP library:
[Errno 101] Network is unreachable.
Please try again.

**Comment by jgyates** [comment]
if you log into your pi via ssh or open a console window on the pi, what happens when you try to ping the SMTP server address like this:

      ping -c 4 smtp.google.com

What is the output of this above command. Note, for this test to be effective you must run it from the pi, not your laptop. You can either log into to the pi via VNC to access the GUI (if you have this setup on your pi) or via ssh.

**Comment by Gadgetmann133** [comment]
I think the root cause of all my issues is my laptop which is a work provided laptop.
I get an access denied error, requires administrative privileges. 
I am going to use my wife's laptop to try get the email working and confirm the test with the Pi.
How do I log into my Pi?  You note running this test must be run from the pi and not my laptop?

**Comment by jgyates** [comment]
you can use ssh to log into the pi over the network. This article will describe how to do that one Mac, Windows or Linux.

https://www.tomshardware.com/how-to/use-ssh-connect-to-remote-computer

for example, MacOS has the ssh program included (windows does not). So from a mac you would type this in a console window:


      ssh pi@192.168.1.10

ssh is the name of the program you are runing
pi is the username on the raspberry pi (assumeing a username of pi)
192.168.1.10 is the IP address of the pi. change this to the IP address of your pi

If you pi is at that IP address you will be prompted for the password for the username you proviced (username in the example was pi).

Once you log in you can type 

     ping -c 4 smtp.google.com

**Comment by Gadgetmann133** [comment]
Update:
Success.   I am now getting emails and text messages from the genmon.
I used my wife's laptop and signed into my gmail account and under security settings selected the app password and generated a password. Copied that into my genmon outbound mail settings, along with the other settings from the instructions and it worked.  
Thanks for your help on this as it has been a journey.
More good news, after rebooting my generator controller, it appears the charging circuit is working again.

**Comment by jgyates** [comment]
great, I am glad it is working for you now.
